app/models/badges/participant_in_12_in_23_badge.rb

```diff
@@ -0,0 +1,21 @@
+module Badges
+  class ParticipantIn12In23Badge < Badge
+    seed "12 in 23",
+      :rare,
+      '12-in-23',
+      'Participated in the #12in23 challenge and completed 5 exercises in a track'
+
+    def award_to?(user)
+      challenge = user.challenges.find_by(challenge_id: User::Challenge::CHALLENGE_12_IN_23)
+      return false if challenge.nil?
+
+      user.solutions.completed.joins(exercise: :track).
+        where('completed_at >= ?', challenge.created_at).
+        group(:track_id).
+        having("count(*) >= 5").
+        exists?
```

Comment:
This is quite slow (1s), due to there not being an appropriate index.

Reply:
Hmm, it's only 5ms when I check...

<img width="644" alt="Screenshot 2023-01-31 at 22 55 23" src="https://user-images.githubusercontent.com/286476/215902513-bc5ff69b-1542-4bf1-ad3a-9e2bca78912b.png">

Reply:
Probably depends on the number of solutions I guess

Reply:
I guess. That's your user above though, who has a lot!

Reply:
True. Ah well